File: tests/test_code/rb/instance_methods/instance_methods.rb
def main
    puts("This is not called")
end

class Abra
    def main
        def nested
            puts("This will reference the main on Abra")
            main
        end
        nested
        puts("Hello from main")
    end

    def main2
        def nested2
            main()
        end
    end
end

a = Abra.new()
a.main()
a.nested()
a.main2()
a.nested2()

File: tests/test_code/php/instance_methods/instance_methods.php
<?php

function main() {
    echo "main; This is called from abra.main";
}

class Abra {
    function main() {
        echo "Abra.main";
        function nested() {
            echo "Abra.nested; This gets placed on Abra";
            main(); # calls global main
        }
        $this->nested();
    }

    function main2() {
        echo "Abra.main2";
        function nested2() {
            echo "Abra.nested2";
            $this.main();
        }
    }
}

echo "A";
$a = new Abra();
echo "B";
$a.main();
echo "C";
$a.nested();
$a.main2();
$a.nested2();

?>
